### PR TITLE
[plugin.video.hbogoeu] 2.1.1

### PR DESCRIPTION
**Pull request by arvvoid**
### Description
Simple, Kodi add-on to access HBO® Go Europe, Spain and Nordic content from Kodi. The add-on is inputstream.adaptive based. Kodi forum thread https://forum.kodi.tv/showthread.php?tid=339798

A valid Hbo Go account in one of the supported regions is required to use the add-on.

The add-on is both Python 2 and 3 compatible (Kodi 19 Python 3 support tested only on macOS and Linux because of pycyptodomex)

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.
- If you see no activity on your PR after a week (so at least one weekend has passed) then please go to the #kodi-dev freenode IRC channel to reach out to the team

### CODE REVIEW

File: plugin.video.hbogoeu/addon.xml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+Important: HBO® Go must be paid for!!! You need a valid HBO® Go account for the add-on to work! Register on the official HBO® Go website for your region
+    </disclaimer>
+    <language>en</language>
+    <platform>all</platform>
+    <license>GPL-2.0</license>
+    <forum>https://forum.kodi.tv/showthread.php?tid=339798</forum>
+    <source>https://github.com/arvvoid/plugin.video.hbogoeu</source>
+    <website>https://arvvoid.github.io/plugin.video.hbogoeu</website>
+    <news>
+v.2.1.0
```
Comment (enen92) [inline plugin.video.hbogoeu/addon.xml]: can you keep this only for the last version please?

File: plugin.video.hbogoeu/hbogolib/base.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    def start(self):
+        country_id = self.addon.getSetting('country_code')
+        country_index = self.country_index(country_id)
+
+        if country_index == -1:
+            self.setup()
+            country_id = self.addon.getSetting('country_code')
+            country_index = self.country_index(country_id)
+            if country_index == -1:
+                xbmcgui.Dialog().ok("ERROR", "Setup failed")
```
Comment (enen92) [inline plugin.video.hbogoeu/hbogolib/base.py]: please localize this strings, they are presented to the user on the interface

File: plugin.video.hbogoeu/hbogolib/base.py [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+                sys.exit()
+
+        if HbogoConstants.countries[country_index][6] == HbogoConstants.HANDLER_EU:
+            from hbogolib.handlereu import HbogoHandler_eu
+            self.handler = HbogoHandler_eu(self.handle, self.base_url, HbogoConstants.countries[country_index])
+        elif HbogoConstants.countries[country_index][6] == HbogoConstants.HANDLER_SPAIN:
+            from hbogolib.handlersp import HbogoHandler_sp
+            self.handler = HbogoHandler_sp(self.handle, self.base_url, HbogoConstants.countries[country_index])
+        else:
+            xbmcgui.Dialog().ok("ERROR", "Unsupported region")
```
Comment (enen92) [inline plugin.video.hbogoeu/hbogolib/base.py]: same here

File: plugin.video.hbogoeu/hbogolib/handlereu.py [hunk 4]
```diff
@@ -0,0 +1,1447 @@
...
+        li_items_list = []
+
+        # 0 - operator name
+        # 1 - operator id
+        # 2 - icon
+        # 3 - is hbogo web or 3th party operator
+        # 4 - login redirection url
+
+        for op_list_item in op_list:
+            li_items_list.append(xbmcgui.ListItem(label=op_list_item[0], iconImage=op_list_item[2]))
```
Comment (enen92) [inline plugin.video.hbogoeu/hbogolib/handlereu.py]: please drop iconImage kwarg

File: plugin.video.hbogoeu/hbogolib/handlereu.py [hunk 5]
```diff
@@ -0,0 +1,1447 @@
...
+                pass  # all is ok media id got from first method ignore
+            except Exception:
+                self.log("Unexpected get media id error: " + traceback.format_exc())
+            liz.addContextMenuItems(items=self.genContextMenu(cid, media_id, HbogoConstants.CONTEXT_MODE_DEFAULT))
+        xbmcplugin.addDirectoryItem(handle=self.handle, url=directory_url, listitem=liz, isFolder=True)
+
+    def addCat(self, name, url, icon, mode):
+        if self.lograwdata:
+            self.log("Adding Cat: " + str(name) + "," + str(url) + "," + str(icon) + " MODE: " + str(mode))
+        category_url = '%s?%s' % (self.base_url, urlencode({
```
Comment (enen92) [inline plugin.video.hbogoeu/hbogolib/handlereu.py]: bad practice to use `file` to name a var

File: plugin.video.hbogoeu/hbogolib/ttml2srt.py [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+        if not self.subtitle['fps']:
+            self.subtitle['fps'] = default_fps
+
+    def write_srt_file(self, str_file, shift=0):
+        filename = str_file + '.srt'
+        if self.subtitle['lang'] is not None:
+            filename = str_file + '.' + self.subtitle['lang'] + '.srt'
+        f = None
+        if (sys.version_info > (3, 0)):
+            f = open(filename, 'w', encoding='utf-8')
```
Comment (enen92) [inline plugin.video.hbogoeu/hbogolib/ttml2srt.py]: xbmcvfs maybe?

File: plugin.video.hbogoeu/hbogolib/uuid_device.py [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+    xbmc.log("[" + str(xbmcaddon.Addon().getAddonInfo('id')) + "]" + msg, xbmc.LOGDEBUG)
+
+
+def get_system_platform():
+    platform = "unknown"
+    if xbmc.getCondVisibility('system.platform.linux') and not xbmc.getCondVisibility('system.platform.android'):
+        platform = "linux"
+    elif xbmc.getCondVisibility('system.platform.linux') and xbmc.getCondVisibility('system.platform.android'):
+        platform = "android"
+    elif xbmc.getCondVisibility('system.platform.xbox'):
```
Comment (enen92) [inline plugin.video.hbogoeu/hbogolib/uuid_device.py]: this is not available in kodi. If you mean the uwp platform use `system.platform.uwp`. See: https://codedocs.xyz/xbmc/xbmc/modules__infolabels_boolean_conditions.html#modules__infolabels_boolean_conditions_System

**Comment by arvvoid** [comment]
Hello @enen92, thank you for your suggestions. I will make the necessary corrections and update this PR later today or in the next few days.

Happy holidays to you too :) ☃️ ❄️ 🎄

File: plugin.video.hbogoeu/hbogolib/uuid_device.py (hunk 7 above)
Comment (arvvoid) [reply plugin.video.hbogoeu/hbogolib/uuid_device.py]: Fixes tnx

File: plugin.video.hbogoeu/hbogolib/ttml2srt.py (hunk 6 above)
Comment (arvvoid) [reply plugin.video.hbogoeu/hbogolib/ttml2srt.py]: I prefer to keep this as it is so I can test it automatically with external tools outside of Kodi. Would the use of xbmcvfs provide an advantage here?

File: plugin.video.hbogoeu/hbogolib/handlereu.py (hunk 5 above)
Comment (arvvoid) [reply plugin.video.hbogoeu/hbogolib/handlereu.py]: Fixed tnx

File: plugin.video.hbogoeu/hbogolib/handlereu.py (hunk 4 above)
Comment (arvvoid) [reply plugin.video.hbogoeu/hbogolib/handlereu.py]: Fixed tnx

File: plugin.video.hbogoeu/addon.xml (hunk 1 above)
Comment (arvvoid) [reply plugin.video.hbogoeu/addon.xml]: Fixed tnx

File: plugin.video.hbogoeu/hbogolib/base.py (hunk 3 above)
Comment (arvvoid) [reply plugin.video.hbogoeu/hbogolib/base.py]: Fixed tnx

File: plugin.video.hbogoeu/hbogolib/base.py (hunk 2 above)
Comment (arvvoid) [reply plugin.video.hbogoeu/hbogolib/base.py]: Fixed tnx

File: plugin.video.hbogoeu/hbogolib/ttml2srt.py (hunk 6 above)
Comment (enen92) [reply plugin.video.hbogoeu/hbogolib/ttml2srt.py]: Only if the file is on network shares since NFS and SMB shares are transparent to xbmcvfs

Comment (arvvoid) [reply plugin.video.hbogoeu/hbogolib/ttml2srt.py]: Ok tnx, the subtitles are read and written to/from the userdata/addon_data/... folder only

**Pull request merged**